### PR TITLE
More pre release fixes

### PR DESCRIPTION
**Pull request by traut**
- doc tweaks
- logging tweaks
- do not print to `stdout` if `--publish` is set
- append `\n` to stdout output

### CODE REVIEW

File: docs/language/configs.md [hunk 1]
```diff
@@ -157,13 +158,16 @@ content text {
...
-Fabric limits which environment variables are exposed inside the templates. By default, only the
-variables that match `FABRIC_*` name glob pattern are available under `env` namespace and in the
-evaluation context. The glob pattern is configured using `expose_env_vars_with_pattern` argument
-in [Global configuration]({{< ref "configs.md#global-configuration">}}).
 
-Name-based filtering minimizes the risk of unintentionally leaking sensitive environment variables to
-untrustworthy plugins. It's recommended to use restrictive glob patterns as much as possible.
+To minimize the risk of unintentionally leaking sensitive environment variables to untrustworthy
+plugins, Fabric limits which environment variables are exposed in the evaluation context.
+
```
Comment (Andrew-Morozko) [inline docs/language/configs.md:164]: I'd change the wording here a little bit, "untrustworthy plugins" sounds quite bad when all of the plugins are written by us 😉 

```suggestion
In accordance to the principle of least privelege, Fabric limits which environment variables are exposed to plugins in the evaluation context.

```

File: docs/language/configs.md [hunk 2]
```diff
@@ -157,13 +158,16 @@ content text {
...
+Only the variables with names that match the pattern set in `expose_env_vars_with_pattern` argument
+(see [Global configuration]({{< ref "configs.md#global-configuration">}})) are available under `.env` key in the evaluation context.
+
+It's recommended to use the most restrictive glob pattern possible.
+
+Note, the filtering doesn't apply to the variables exposed in `env` object.
```
Comment (Andrew-Morozko) [inline docs/language/configs.md:170]: Perhaps clarify why the difference is there?
```suggestion
Note, that the filtering doesn't apply to the variables exposed in `env` object: it's the user's responsibility to make sure that their Fabric template doesn't expose sensitive environment variables.
```

File: docs/language/context.md [hunk 3]
```diff
@@ -71,13 +71,13 @@ content text {
...
+content block will render `Hello, World!`.
 
 ### Inheritance
 
-Variables defined in a parent block (such as `document` or `section`) are available for use in nested
-blocks. Nested blocks can redefine a variable in their context, overwriting the parent variable's
-value.
+Variables defined in a parent block (such as `document` or `section`) are available for use in
+nested blocks. Note that nested blocks can redefine a variable in their context, overwriting the
+parent variable's value.
```
Comment (Andrew-Morozko) [inline docs/language/context.md:80]: I'd reserve "overwriting" for the ref-vars situation, and use shadowing in the nesting situation, it seems more intuitive.
```suggestion
nested blocks. Note that nested blocks can redefine a variable in their context, shadowing the
parent variable's value inside their scope.
```

File: docs/tutorial.md [hunk 4]
```diff
@@ -217,21 +221,31 @@ Mar 11 19:20:10.787 INF Installing plugin name=blackstork/openai version=0.4.0
...
+OpenAI API requires API key for authentication. The key must be set in [`openai_text`]({{< ref
+"plugins/openai/content-providers/openai_text" >}}) provider's configuration block. It's recommended
+to store credentials separately from Fabric code, and use the `env` object (see [Environment variables]({{< ref
+"configs.md#environment-variables" >}})).
+
+We can specify OpenAI API key in `OPENAI_API_KEY` environment variable and access it in Fabric file
+with `env.OPENAI_API_KEY`.
 
-The `config` block for the `openai_text` content provider would look like this:
+The `config` block for the `openai_text` content provider would looks like this:
```
Comment (Andrew-Morozko) [inline docs/tutorial.md:241]:
```suggestion
The `config` block for the `openai_text` content provider looks like this:

```

**Review by Andrew-Morozko — APPROVED** [review]
LGTM, just some suggestions and nitpicks

File: docs/language/configs.md (hunk 1 above)
Comment (traut) [reply docs/language/configs.md:164]: good point!

File: docs/language/configs.md [hunk 5]
```diff
@@ -28,10 +28,10 @@ Within the codebase, only one `fabric` block can be defined.
...
   SemVer (refer to Terraform [version constraint
   syntax](https://developer.hashicorp.com/terraform/language/expressions/version-constraints#version-constraint-syntax)).
-- `expose_env_vars_with_pattern`: (optional) a glob pattern for filtering the environment variables
-  exposed in the templates by name. Default value is `FABRIC_*`. To expose all environment
-  variables, set the value to `*`. See [Environment variables]({{< ref
-  "configs.md#environment-variables">}}) for the details.
+- `expose_env_vars_with_pattern`: (optional) a glob pattern for environment variable names. Only the
+  environment variables that match the pattern are exposed in the evaluation context. Default value
+  is `FABRIC_*`. To expose all environment variables, set the value to `*`. See [Environment
+  variables]({{< ref "configs.md#environment-variables">}}) for the details.
```
Comment (Andrew-Morozko) [inline docs/language/configs.md:34]: FYI: Setting to null or an empty string would expose no environment variables

**Pull request merged**